pkg/cmd/cli/schedule/delete.go

```diff
@@ -19,29 +19,149 @@ package schedule
 import (
 	"fmt"
 
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	kubeerrs "k8s.io/apimachinery/pkg/util/errors"
+
+	arkv1api "github.com/heptio/ark/pkg/apis/ark/v1"
 	"github.com/heptio/ark/pkg/client"
 	"github.com/heptio/ark/pkg/cmd"
+	"github.com/heptio/ark/pkg/cmd/cli"
+	"github.com/heptio/ark/pkg/cmd/util/flag"
+	clientset "github.com/heptio/ark/pkg/generated/clientset/versioned"
 )
 
+// NewDeleteCommand creates and returns a new cobra command for deleting schedules.
 func NewDeleteCommand(f client.Factory, use string) *cobra.Command {
+	o := &DeleteOptions{}
 	c := &cobra.Command{
-		Use:   fmt.Sprintf("%s NAME", use),
-		Short: "Delete a schedule",
-		Args:  cobra.ExactArgs(1),
+		Use:   fmt.Sprintf("%s [NAMES]", use),
+		Short: "Delete schedules",
+		Example: `	# delete a schedule named "schedule-1"
+		ark schedule delete schedule-1
+	
+		# delete a schedule named "schedule-1" without prompting for confirmation
+		ark schedule delete schedule-1 --confirm
+	
+		# delete schedules named "schedule-1" and "schedule-2"
+		ark schedule delete schedule-1 schedule-2
+	
+		# delete all schedules labelled with foo=bar"
+		ark schedule delete --selector foo=bar
+		
+		# delete all schedules
+		ark schedule delete --all`,
+
 		Run: func(c *cobra.Command, args []string) {
-			arkClient, err := f.Client()
-			cmd.CheckError(err)
+			cmd.CheckError(o.Complete(f, args))
+			cmd.CheckError(o.Validate(c, f, args))
+			cmd.CheckError(o.Run())
+		},
+	}
 
-			name := args[0]
+	o.BindFlags(c.Flags())
+	return c
+}
 
-			err = arkClient.ArkV1().Schedules(f.Namespace()).Delete(name, nil)
-			cmd.CheckError(err)
+// DeleteOptions specifies params used for deleting a schedule.
+type DeleteOptions struct {
```

Comment:
This struct is common across all three commands

Reply:
Would it make sense to extract these in a common place?

Reply:
@shubheksha Thanks, I was just looking and saw the repetition.

I'm not opposed to putting the `DeleteOptions` type and its methods into a central location. At first I thought `pkg/cmd/cli/delete` might be a good place for it, but I think that would create circular dependencies, as it imports these files to provide the same functions, just with reversed names.

Perhaps `pkg/cmd/cli/types`?

Reply:
SGTM!
What about the `Complete()` and `Validate()` methods? AFAICT, they are identical. Would it make sense to move those to a common place too?

Reply:
@shubheksha Since the methods are on that type, moving them to the same file makes sense to me, but I'm not sure if that's a go antipattern. @carlisia what do you think?

Reply:
Ping @carlisia 
I wasn't sure if it's an anti-pattern either. How does one decide if it's one or not? 😄

Reply:
Hi, I missed the ping, sorry! Not an anti-pattern, it's exactly what I would do (keep the types + methods in the same file). How about we add them to this file: https://github.com/heptio/ark/blob/master/pkg/cmd/cli/common.go